testgrid/cmd/configurator/prow_test.go

```diff
@@ -0,0 +1,567 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	prowConfig "k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/testgrid/config"
+	"reflect"
+	"testing"
+)
+
+const ProwDefaultGCSPath = "pathPrefix/"
+const ProwJobName = "TestJob"
+
+func Test_applySingleProwjobAnnotations(t *testing.T) {
+	tests := []struct {
+		name           string
+		initialConfig  *config.Configuration
+		prowJobType    prowapi.ProwJobType
+		annotations    map[string]string
```

Comment:
Specify all fields for tests, even the empty/default ones?

Reply:
I'd argue that including empty values is redundant clutter in most cases.

In addition, since we're usually dealing with pointers, there's a behavioral difference between `expectedConfig: &config.Configuration{}` and `expectedConfig: nil`. The former indicates "I'm expecting a valid, but empty configuration", while the latter indicates "I'm expecting something to go wrong". I think omitting the line completely conveys "this should fail" a bit clearer.